zipline/testing/predicates.py

```diff
@@ -84,7 +84,37 @@ def __ne__(other):
 
     def __repr__(self):
         return '<%s>' % type(self).__name__
-    __str__ = __repr__
+
+
+class instance_of(object):
+    """An object that compares equal to any instance of a given type or types.
+
+    Parameters
+    ----------
+    types : type or tuple[type]
+        The types to compare equal to.
+    exact : bool, optional
+        Only compare equal to exact instances, not instances of subclasses?
+    """
+    def __init__(self, types, exact=False):
+        if not isinstance(types, tuple):
+            types = (types,)
+
+        self.types = types
+        self.exact = exact
+
+    def __eq__(self, other):
+        if self.exact:
+            return type(other) in self.types
+
+        return isinstance(other, self.types)
+
+    def __repr__(self):
+        return '<%s: %r%s>' % (
+            type(self).__name__,
+            self.types[0] if len(self.types) == 1 else self.types,
```

Comment:
This is going to crash on the `len` call if `types` is a single type.

Reply:
Oh, nvm, you're converting at construction time.

Reply:
it is always a tuple, check __init__